### PR TITLE
new inject creative api

### PR DESCRIPTION
**Pull request by webb04**
Requires iOS and Android changes to be merged at the same time.

### CODE REVIEW

**Comment by alexanderedge** [comment]
I see your comment on the JIRA ticket:

> We might need to check how impressions are tracked because we no longer have an id sent to the webView.

Would it be possible to pass the ID into `injectCreative(...)`?

**Comment by webb04** [comment]
yep we can have another `ID` parameter and use it how we do currently e.g. `signalDevice("creative_impression/ID")`

**Comment by webb04** [comment]
@alexanderedge I remember @SLambrakis saying we only have one Epic per page so iOS/Android would know what that ID was?

Happy to add it back if it's better to handle native side though.

**Comment by alexanderedge** [comment]
> @alexanderedge I remember @SLambrakis saying we only have one Epic per page so iOS/Android would know what that ID was?
> 
> Happy to add it back if it's better to handle native side though.

Ah, okay. I was thinking of minimising changes iOS-side. I'll take a look tomorrow and see if we can get around not sending the ID into the webview.

File: ArticleTemplates/assets/js/modules/creativeInjector.js [hunk 1]
```diff
@@ -67,41 +51,13 @@ function injectEpicCreative(creativeContainer) {
...
-
-    creativeContainer.id = id;
-    creativeContainer.classList.add('creative-container');
-    creativeContainer.classList.add(`${type}-creative-container`);
-    creativeContainer.innerHTML = html;
-
-    if (type === 'inline-article') {
-        injectInlineCreative(creativeContainer, 10);
-    } else {
+function injectCreative(title, body, firstButton, secondButton) {
```
Comment (alexanderedge) [inline ArticleTemplates/assets/js/modules/creativeInjector.js]: I was speaking with @SLambrakis and she mentioned that we only inject epics. Therefore, should this function be called `injectEpic(...)` to make it clear what it does?

Comment (webb04) [reply ArticleTemplates/assets/js/modules/creativeInjector.js]: yep can do, will be clear it's a new api too.

**Comment by alexanderedge** [comment]
Unsure what the button title length requirements are but the arrow isn't vertically centred if it wraps:

Simulator Screen Shot - iPhone Xʀ - 2019-09-18 at 11 32 05

**Comment by alexanderedge** [comment]
> @alexanderedge I remember @SLambrakis saying we only have one Epic per page so iOS/Android would know what that ID was?
> 
> Happy to add it back if it's better to handle native side though.

No need to add the ID back as there will only be a single creative.

**Comment by webb04** [comment]
I'll remove the max-width on that button and change `injectCreative` to `injectEpic`

**Comment by webb04** [comment]
Screen Shot 2019-09-18 at 12 35 26

File: ArticleTemplates/assets/js/modules/creativeInjector.js [hunk 2]
```diff
@@ -5,58 +5,42 @@ import {
...
-    if (!trackedImpressions.includes(id) && isElementPartiallyInViewport(creativeContainer)) {
-        signalDevice(messageName);
-        trackedImpressions.push(id);
-    }
+let impressionSeen = false;
+
+function buttonHtml(buttonText) {
+    return `
+        <div>
+            <a class="epic-button" href="x-gu://creative_tap/premium">
```
Comment (JamieB-gu) [inline ArticleTemplates/assets/js/modules/creativeInjector.js]: Will this URL will be hit for both buttons on Android?

File: ArticleTemplates/assets/js/modules/creativeInjector.js [hunk 3]
```diff
@@ -5,58 +5,42 @@ import {
...
-    let i;
-    let liveBlogEpicContainer;
-
-    for (i = 0; i < liveBlogEpicContainers.length; i++) {
-        liveBlogEpicContainer = liveBlogEpicContainers[i];
-        liveBlogEpicContainerId = liveBlogEpicContainer.getAttribute('id');
-
-        liveBlogEpicContainer.setAttribute('data-tracked', 'true');
+function isCreativeInView(creativeContainer) {
+    const messageName = `creative_impression`;
```
Comment (JamieB-gu) [inline ArticleTemplates/assets/js/modules/creativeInjector.js]: Could this be a normal string?

File: ArticleTemplates/assets/scss/modules/_injectedCreative.scss [hunk 4]
```diff
@@ -1,19 +1,46 @@
...
+        color: color(brightness-7);
+        background: color(tone-highlight);
+        border-radius: 100px;
+        margin: 0 15px 15px 0;
+        padding: 6px 6px 4px 15px;
+        font-size: 16px;
+        font-weight: 900;
+        font-family: "Guardian Text Sans Web", "Helvetica Neue", Helvetica, Arial, "Lucida Grande", sans-serif;
+        line-height: 26px;
+    }
```
Comment (JamieB-gu) [inline ArticleTemplates/assets/scss/modules/_injectedCreative.scss]: As discussed offline, font re-ordering.

**Pull request merged**